relations/src/account.rs

```diff
@@ -0,0 +1,22 @@
+use halo2_base::{gates::GateChip, utils::BigPrimeField, AssignedValue, Context};
+
+use crate::operation::{CircuitOperation, Operation};
+
+pub trait Account<F: BigPrimeField> {
+    type CircuitAccount: CircuitAccount<F>;
+    type Op: Operation<F>;
+
+    fn update(&self, op: Self::Op) -> Self;
+
+    fn to_array(&self) -> Vec<F>;
```

Comment:
nit: I would introduce a trait (or, if we are ok with consuming, then simply use `Into<Vec<F>>`) for this conversion - it seems to be common for a few types

applies also to the circuit- version as well

Reply:
Done